gossip/hub/handlers_test.go

```diff
@@ -224,9 +239,31 @@ func makeGetSLRRsp(t *testing.T, timestamp, treeSize int64, hash []byte) *trilli
 	return &trillian.GetLatestSignedLogRootResponse{SignedLogRoot: makeSLR(t, timestamp, treeSize, hash)}
 }
 
+// signData returns data, signature-over-data as a 2 element slice of base-64 encoded strings.
+func signData(t *testing.T, data []byte) []interface{} {
+	digest := sha256.Sum256(data)
+	sig, err := testSourceKey.Sign(rand.Reader, digest[:], crypto.SHA256)
+	if err != nil {
+		t.Fatalf("Failed to sign data: %v", err)
+	}
+	return []interface{}{base64.StdEncoding.EncodeToString(data), base64.StdEncoding.EncodeToString(sig)}
+}
+
 func TestAddSignedBlob(t *testing.T) {
 	ctx := context.Background()
-	data := []byte{0x00, 0x00, 0x00}
+
+	th := ct.TreeHeadSignature{
+		Version:       ct.V1,
+		SignatureType: ct.TreeHashSignatureType,
+		Timestamp:     1000000,
+		TreeSize:      100,
+	}
+	rand.Read(th.SHA256RootHash[:])
```

Comment:
Might be better to have a static `TreeHeadSignature` for repeatable tests.

Reply:
There's already a bunch of randomly-generated data (cf. `testSourceKey`) in the tests, so the tests are already non byte-for-byte repeatable.

(We've used hard-coded keys and signatures in the past, but I've always disliked their oracular nature and lack of provenance.  For the gossip stuff I've been taking a different approach and trying to generate keys on the fly throughout  -- and I think it's made life easier (and more realistic) as a result.)